### PR TITLE
Fix combo of props and getStaticPaths

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Fixes edge case where Props generic detection was getting tripped up.
- Runs a few tests that I'm not sure were running previously

## Testing

Adds a new test

## Docs

Bug fix only

### CODE REVIEW

File: packages/compiler/test/tsx/non-latin.ts [hunk 1]
```diff
@@ -66,3 +66,5 @@ export default function __AstroComponent_(_props: Record<string, any>): any {}`;
   const { code } = await convertToTSX(input, { sourcemap: 'external' });
   assert.snapshot(code, output, `expected code to match snapshot`);
 });
+
+test.run();
```
Comment (jasikpark) [inline packages/compiler/test/tsx/non-latin.ts:70]: is this a requirement of uvu tests ig?

File: internal/js_scanner/js_scanner.go [hunk 2]
```diff
@@ -339,7 +339,7 @@ outer:
 		// Track our current position
 		i += len(value)
 	}
-	if len(genericsIdents) > 0 && ident != defaultPropType {
+	if start > 0 && len(genericsIdents) > 0 && ident != defaultPropType {
```
Comment (jasikpark) [inline internal/js_scanner/js_scanner.go:342]: heh? is this just saying that `Props` is a magic name

Comment (natemoo-re) [reply internal/js_scanner/js_scanner.go:342]: Yep, `Props` is a magic name. In this case our lexing was getting tripped up because of the `Record<string, never>` and it was injecting the whole `---
type Props = Record<string, never>` string in the wrong place.

File: packages/compiler/test/tsx/non-latin.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/compiler/test/tsx/non-latin.ts:70]: Yeah, I'm not sure if these were still running, but better safe than sorry!

**Review by Princesseuh — APPROVED** [review]

**Review by jasikpark — APPROVED** [review]
